contracts/libraries/HyperdriveExecution.sol

```diff
@@ -0,0 +1,578 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.20;
+
+import { IHyperdrive } from "hyperdrive/contracts/src/interfaces/IHyperdrive.sol";
+import { FixedPointMath } from "hyperdrive/contracts/src/libraries/FixedPointMath.sol";
+import { HyperdriveMath } from "hyperdrive/contracts/src/libraries/HyperdriveMath.sol";
+import { SafeCast } from "hyperdrive/contracts/src/libraries/SafeCast.sol";
+import { YieldSpaceMath } from "hyperdrive/contracts/src/libraries/YieldSpaceMath.sol";
+import { Packing } from "openzeppelin/utils/Packing.sol";
+import { IEverlong } from "../interfaces/IEverlong.sol";
+import { IEverlongEvents } from "../interfaces/IEverlongEvents.sol";
+
+// TODO: Extract into its own library.
+uint256 constant HYPERDRIVE_SHARE_RESERVES_BOND_RESERVES_SLOT = 2;
+uint256 constant HYPERDRIVE_LONG_EXPOSURE_LONGS_OUTSTANDING_SLOT = 3;
+uint256 constant HYPERDRIVE_SHARE_ADJUSTMENT_SHORTS_OUTSTANDING_SLOT = 4;
+
+/// @author DELV
+/// @title HyperdriveExecutionLibrary
+/// @notice Library to handle the execution of trades with hyperdrive.
+/// @custom:disclaimer The language used in this code is for coding convenience
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+library HyperdriveExecutionLibrary {
```

Comment:
Nit: Just as a general comment, some of these functions are prefixed with a leading underscore, and some aren't. We typically avoid the leading underscore in libraries. Any reason to include them here?

Reply:
The meaning is consistent with what an underscore indicates in contracts, that the function is not to be called by external parties and is only used internally.

If this is against the desired style, is there another preferred method for indicating the same?

Reply:
You should be able to make it a private function.

Reply:
That said, it’s not a big deal. I’m asking because it’s not what we usually do, but your reasoning makes sense